File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// https://leetcode-cn.com/problems/two-sum/
func twoSum(nums []int, target int) []int {
	ret := make([]int, 2)
	numsLen := len(nums)
	
	if numsLen == 0 {
		return ret
	}

	indexs := make(map[int]int)

	for i := 0; i < numsLen; i++ {
		indexs[nums[i]] = i
	}

	for i := 0; i < numsLen; i ++ {
		num1 := nums[i]
		num2 := target - num1

		if _, ok := indexs[num2]; ok {
			index1 := i
			index2 := indexs[num2]

			if index1 == index2 {
				continue
			}

			ret[0] = index1
			ret[1] = index2
			return ret
		}
	}

	return ret
}

File: 1.两数之和.swift
/*
 * @lc app=leetcode.cn id=1 lang=swift
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        if nums == nil {
            return [Int]()
        }

        var recordMap : [Int : Int] = [Int : Int]()
        for (index, value) in nums.enumerated() {
            let missValue = target - value
            if recordMap.keys.contains(missValue) {
                return [index, recordMap[missValue]!]
            }
            recordMap[value] = index
        }

        return [Int]()
    }
}
// @lc code=end

